### ISSUE TITLE
Validate during training

### ISSUE THREAD

**Issue by kleingeo** [ultralytics/ultralytics#22677]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

This is a very basic question, but I cannot seem to find the answer.

I am training a `Yolo11n-seg` with a custom dataset. The training works (or at least seems to work) fine. 

```yaml
path: custom_data # dataset root dir
train: images/train # train images (relative to 'path') 128 images
val: images/val # val images (relative to 'path') 128 images
test: images/val

names:
 0. [class 0]
 1. [class 1]
 2. [class 2]
 ...
```

```python
from ultralytics import YOLO
from utils.plots import plot_results

if __name__ == '__main__':

    # Load a COCO-pretrained YOLO11n model
    model = YOLO("yolo11n-seg.pt")
    
    results = model.train(data="custom_data.yaml", epochs=1000, imgsz=640, save=True, save_period=10, batch=16, project='custom_data', exist_ok=True, plots=True, val=True)
```

It is all very basic and it seems to be training correctly. However, the `plots=True` argument for training does not seem to be properly creating the visualizations as specified. Is there a different agument so that validation examples can be saved during training, or am I missing something completely? The documentation is not clear about this.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @kleingeo, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get started quickly—an Ultralytics engineer will follow up shortly.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including:
-Exact code/CLI command used and full logs
-Your `ultralytics` version and environment (OS, Python, PyTorch, CUDA)
-Example images from your dataset and your dataset YAML
-Any generated artifacts (e.g., `results.png`, `runs/` contents) or screenshots
And please verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a Python>=3.8 environment with PyTorch>=1.8 to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):

- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- Docker Image. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

To help us assist you faster regarding validation visuals during training with YOLO11 segmentation, please share:
-Your exact `model.train(...)` call and any `model.val(...)` usage
-The `runs/` directory contents (e.g., `results.png`, `val_batch*.jpg`, `confusion_matrix.png`) or screenshots
-Where you expect plots to appear and what’s missing
We’ll take it from there! 💡

**Comment by Y-T-G** [comment]
The visualization is only created during the final validation at the end of the training. If you want it to appear during training too, you can use a callback:

```python
def enable_plot(validator):
    validator.args.plots = True

model.add_callback("on_val_start", enable_plot)
```

However, it will overwrite the old plots each epoch.

**Comment by kleingeo** [comment]
Thanks, that works. Having it overwrite should be fine for now, I'm just doing to initial testing.

Is this stuff writen anywhere in the documentation? Just curious for my own understanding.

**Comment by Y-T-G** [comment]
No. It's a custom callback. If you understand the code, you can modify the behavior using callbacks like this
